### PR TITLE
Write logs

### PR DESCRIPTION
**Pull request by juliasaveliff**
Logs done for:
- [x] Assets
- [x] Models

To do (in separate PR):
- [ ] Users
- [ ] Datacenters
- [ ] Racks
- [ ] Network changes
- [ ] Power changes

### CODE REVIEW

File: rackcity/models/log.py [hunk 1]
```diff
@@ -17,9 +17,11 @@ class Log(models.Model):
         verbose_name="related user",
         null=True,
     )
-    related_assets = models.ManyToManyField(
+    related_asset = models.ForeignKey(
```
Comment (juliasaveliff) [inline rackcity/models/log.py]: per our discussion earlier today, we are only going to link ONE asset to any log (if a change involves multiple assets such as network changes, then multiple logs will be written, one for each affected asset)

File: rackcity/utils/log_utils.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    elif isinstance(element, ITModel):
+        element_type = ElementType.MODEL.value
+        element_name = element.vendor + " " + element.model_number
+        related_model = element
+    elif isinstance(element, User):
+        element_type = ElementType.USER.value
+        element_name = element.username
+    elif isinstance(element, Datacenter):
+        element_type = ElementType.DATACENTER.value
+        element_name = element.name
```
Comment (benauriemma) [inline rackcity/utils/log_utils.py]: Picky but I might prefer abbreviation here?  Would approve either way though

File: rackcity/utils/log_utils.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        date=date,
+        log_content=log_content,
+        user=user,
+        related_asset=related_asset,
+        related_model=related_model,
+    )
+    log.save()
+
+
+def log_delete(user, element_type, element_name):
```
Comment (benauriemma) [inline rackcity/utils/log_utils.py]: I thought this was `delete_log` for a sec and got scared. Phew

File: rackcity/utils/log_utils.py (hunk 2 above)
Comment (juliasaveliff) [reply rackcity/utils/log_utils.py]: i can dig it

File: rackcity/utils/log_utils.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    """
+    Specified action should be Action enum, element_type should be ElementType
```
Comment (benauriemma) [inline rackcity/utils/log_utils.py]: Think this comment is stale from above

File: rackcity/utils/log_utils.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+            related_element.vendor,
+            related_element.model_number
+        ])
+        related_model = related_element
+    elif isinstance(related_element, User):
+        element_type = ElementType.USER.value
+        element_name = related_element.username
+    elif isinstance(related_element, Datacenter):
+        element_type = ElementType.DATACENTER.value
+        element_name = related_element.abbreviation
```
Comment (juliasaveliff) [inline rackcity/utils/log_utils.py]: @benauriemma changed to abbreviation

File: rackcity/utils/log_utils.py (hunk 3 above)
Comment (juliasaveliff) [reply rackcity/utils/log_utils.py]: i would LITERALLY never

File: rackcity/views/asset_views.py [hunk 6]
```diff
@@ -319,13 +327,14 @@ def asset_modify(request):
                 asset_id=existing_asset.id
             )
         except MacAddressException as error:
-            failure_message += "Some mac addresses couldn't be saved. " + \
+            failure_message = "Some mac addresses couldn't be saved. " + \
```
Comment (juliasaveliff) [inline rackcity/views/asset_views.py]: this was `+=` before from your recent PR but failure_message isn't actually defined yet

Comment (benauriemma) [reply rackcity/views/asset_views.py]: Ooh good catch thank you

**Pull request merged**